File: Python/sentimental-readability/readability.py
from cs50 import get_string
import math


def main():
    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    index = 0  # formula
    l = 0  # average number of letter per 100 words
    s = 0  # average number of sentences per 100 words.

    l = (letters / words) * 100
    print("", l)
    s = (sentences / words) * 100
    print("", s)

    index = (0.0588 * l) - (0.296 * s) - 15.8
    print("", index)

    if (math.ceil(index)) >= 16:
        print("Grade 16+")
    elif math.ceil(index) < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {round(index)}")


def count_letters(text):
    c_words = 0  # we set a counter for the number of words
    i = 0  # we set a external iterator so it can index the string
    for j in text:  # this is a for loop, the a iterator will be added plus 1 automatically, this doesn't interfer with the i iterator
        if (text[i] >= 'a' and text[i] <= 'z') or (text[i] >= 'A' and text[i] <= 'Z'):
            c_words += 1
        i += 1
    print(f"Number of letters : {c_words}")
    return c_words


def count_words(text):
    s_words = 1
    i = 0
    for j in text:
        if text[i] == " ":
            s_words += 1
        i += 1
    print(f"Number of words: {s_words}")
    return s_words


def count_sentences(text):
    sent_w = 0
    i = 0
    for j in text:
        if (text[i] == '.' or text[i] == '?' or text[i] == '!'):
            sent_w += 1
        i += 1
    print(f"Number of sentences: {sent_w}")
    return sent_w


main()

File: C_lang/readability_old/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text:  ");
    int letters = count_letters(text);
    // printf("letters: %i\n", letters);
    int words = count_words(text);
    // printf("words: %i\n", words);
    int sentences = count_sentences(text);
    // printf("sentences:  %i \n", sentences);

    double avl = ( (double)letters / (double)words) * 100;
    // printf("%f \n", avl);
    double avs = ((double)sentences / (double)words ) * 100;
    double index = (0.0588 * avl) - (0.296 * avs) - 15.8;


    if (index < 1.0 )
    {
        printf("Before Grade 1\n");
    }
    else if ((int) round(index) >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    printf("Grade %i\n", (int) round(index));


}

int count_letters(string text)
{
    int i = 0;
    int j = 0;
    while (text[i] != '\0')
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            j++;
        }
        i++;

    }
    return j;
}

int count_words(string text)
{
    int k = 0;
    int l = 1;
    while(text[k] != '\0')
    {
        if (text[k] == ' ')
        {
            l++;
        }
        k++;
    }
     return l;
}

int count_sentences(string text)
{
    int m = 0;
    int o = 0;
    while (text[m] != '\0')
    {
        if (text[m] == '.' || text[m] == '?' || text[m] == '!')
        {
            o++;
        }
        m++;
    }
    return o;
}